app/Controllers/Http/AuthController.ts

```diff
@@ -0,0 +1,52 @@
+import type { HttpContextContract } from "@ioc:Adonis/Core/HttpContext";
+import User from "App/Models/User";
+import { schema, rules } from "@ioc:Adonis/Core/Validator";
+
+const userSchema = schema.create({
+  email: schema.string({}, [rules.email()]),
+  password: schema.string({}, [
+    rules.minLength(8),
+    rules.maxLength(16),
+    rules.regex(/[a-zA-Z0-9]+/),
+  ]),
+});
+export default class AuthController {
+  public async login({ auth, request, response }) {
+    try {
+      const payload = await request.validate({ schema: userSchema });
+      const user: User | null = await User.findBy("email", payload.email);
```

Comment:
```suggestion
      const user = await User.findBy("email", payload.email);
```
No need to explicitly mention the type here